doc/modules/decomposition.rst

```diff
@@ -278,6 +262,100 @@ factorization, while larger values shrink many coefficients to zero.
      R. Jenatton, G. Obozinski, F. Bach, 2009
 
 
+.. _kernel_PCA:
+
+Kernel Principal Component Analysis (kPCA)
+==========================================
+
+Exact Kernel PCA
+----------------
+
+:class:`KernelPCA` is an extension of PCA which achieves non-linear
+dimensionality reduction through the use of kernels (see :ref:`metrics`). It
+has many applications including denoising, compression and structured
+prediction (kernel dependency estimation). :class:`KernelPCA` supports both
+``transform`` and ``inverse_transform``.
+
+.. figure:: ../auto_examples/decomposition/images/sphx_glr_plot_kernel_pca_001.png
+    :target: ../auto_examples/decomposition/plot_kernel_pca.html
+    :align: center
+    :scale: 75%
+
+.. topic:: Examples:
+
+    * :ref:`sphx_glr_auto_examples_decomposition_plot_kernel_pca.py`
+
+.. topic:: References:
+
+    * Kernel PCA was introduced in "Kernel principal component analysis"
+      Bernhard Schoelkopf, Alexander J. Smola, and Klaus-Robert Mueller. 1999.
+      In Advances in kernel methods, MIT Press, Cambridge, MA, USA 327-352.
+
+
+.. _kPCA_Solvers:
+
+Choice of solver for Kernel PCA
+-------------------------------
+
+While in :class:`PCA` the number of components is bounded by the number of
+features, in :class:`KernelPCA` the number of components is bounded by the
+number of samples. Many real-world datasets have large number of samples! In
+these cases finding *all* the components with a full kPCA is a waste of
+computation time, as data is mostly described by the first few components
+(e.g. ``n_components<=100``).
+
+The optional parameter ``eigen_solver='randomized'`` can be used to
+*significantly* reduce the computation time when the number of requested
+``n_components`` is small compared with the number of samples.
+This is done automatically by default if you do not specify any
+``eigen_solver`` and if the required ``n_components`` is less than 80% of
+the number of samples, see :class:`KernelPCA` for details.
+
+The time complexity of the randomized :class:`KernelPCA` is
+:math:`O(n_{\mathrm{samples}}^2 \cdot n_{\mathrm{components}})`
+instead of :math:`O(n_{\mathrm{samples}}^3)` for the exact method
+implemented with ``eigen_solver='dense'``.
+
+The memory footprint of randomized :class:`KernelPCA` is also proportional to
+:math:`2 \cdot n_{\mathrm{samples}} \cdot n_{\mathrm{components}}` instead of
```

Comment:
Very nit: Do we include the `2` here when comparing this to `n_samples^2`?

Reply:
Frankly I can not answer easily here, the details are quite far away now :)

Reply:
Now I remember: this is a copy of the documentation in the [randomized PCA](https://scikit-learn.org/stable/modules/decomposition.html#pca-using-randomized-svd) that is already in the sklearn doc:

![image](https://user-images.githubusercontent.com/3236794/99870462-59d29980-2bd3-11eb-82dd-7cb4757b0e7d.png)

Reply:
But honestly I am not sure (even for this original comment in PCA doc, I don't know who wrote it ) that this comment is relative to usage of the "randomized" solver, or simply to the fact that when `n_components` is less than the max, then only this number of support vectors have to be kept in memory. We should maybe contact the original author of the PCA doc to understand why he/she wrote this ? As for KernelPCA, I'll just align.